File: src/main/java/info/skarby/examples/example6/StaticMembersExample.java
package info.skarby.examples.example6;

public class StaticMembersExample {
    public static void main(String[] args) {
        Person john = Person.fromFullName("John P. Doe");
        System.out.println("First name(s): " + john.getFirstName() + ", last name: " + john.getLastName());
    }
}

class Person {
    private final String firstName;
    private final String lastName;

    public Person(String firstName, String lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public static Person fromFullName(String fullName) {
        String firstName = fullName.substring(0, fullName.lastIndexOf(' '));
        String lastName = fullName.substring(fullName.lastIndexOf(' ') + 1);
        return new Person(firstName, lastName);
    }
}

File: src/main/kotlin/info/skarby/examples/example6/StaticMembersExample.kt
package info.skarby.examples.example6

fun main(args: Array<String>) {
    val john = KPerson.fromFullName("John P. Doe")
    println("First name(s): ${john.firstName}, last name: ${john.lastName}")
}

data class KPerson(val firstName: String, val lastName: String) {
    companion object {
        fun fromFullName(fullName: String): KPerson {
            val firstName = fullName.substringBeforeLast(' ')
            val lastName = fullName.substringAfterLast(' ')
            return KPerson(firstName, lastName)
        }
    }
}